style-guide.md

```diff
@@ -546,3 +546,55 @@ def long_signature(
         thing_d=None):
     pass
 ```
+
+# Typing
+
+Since all new code is required to come with type hints ([PEP 484](https://www.python.org/dev/peps/pep-0484/)), the following aims to provide answers to some common questions around working with types and `mypy`.
+
+## Choosing the right type
+
+When working with types, we often have to choose between multiple different types that all may seem like the right candidate for the code in question.
+
+### Function variance
+
+As a rule of thumb, the **parameters** of a function **should** be typed to accept the *most flexible* type that the function can work with.
+
+```python
+# good
+def triple_len(seq: Sequence[T]) -> int:
+    return len(seq) * 3
+
+res_1 = triple_len(["some", "irrelevant", "data"])
+res_2 = triple_len(("some", "irrelevant", "data",))
+
+# bad
+def triple_len(seq: List[T]) -> int:
+    return len(seq) * 3
+
+res_1 = triple_len(["some", "irrelevant", "data"])
```

Comment:
Looks like you may have intended this to be a tuple rather than a list.

Reply:
This is intentional. The `Tuple` case is demoed (but commented) further down as it doesn't work if the parameter of `triple_len` is set as `List[T]`, hence we want it as `Sequence[T]` which works with both `Tuple` and `List`